core/src/main/java/com/sun/mail/util/PropUtil.java

```diff
@@ -149,6 +164,16 @@ private static int getInt(Object value, int def) {
 	return def;
     }
 
+	private static ScheduledExecutorService getScheduledExecutorService(Object value) {
+		if (value == null)
+			return null;
+		if (value instanceof ScheduledExecutorService) {
+			return (ScheduledExecutorService) value;
+		}
+
+		throw new ClassCastException("Wrong class of scheduled executor");
```

Comment:
Given the way this method works we could just reduce it to `return (ScheduledExecutorService) value;` or `ScheduledExecutorService.class.cast(value)`.  I've found that the latter is a bit lazier with the class verifier.  Both will have more detailed messages about the types at play versus what we have.

Reply:
fixed